pkg/pop/msal_confidential.go

```diff
@@ -0,0 +1,32 @@
+package pop
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/kubelogin/pkg/internal/pop"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+)
+
+func AcquirePoPTokenConfidential(
+	context context.Context,
+	popClaims map[string]string,
+	scopes []string,
+	cred confidential.Credential,
+	authority,
+	clientID,
+	tenantID string,
+	options *azcore.ClientOptions,
+	popKeyFunc func() (*SwKey, error),
+) (string, int64, error) {
+
+	internalPopKeyFunc := func() (*pop.SwKey, error) {
```

Comment:
let's get rid of this and just pass the `popKeyFunc` directly to `AcquirePoPTokenConfidential`